### ISSUE TITLE
[vcpkg-tool] 2024-09-18 Release is broken

### ISSUE THREAD

**Issue by Neumann-A** [microsoft/vcpkg#41235]
[config-x64-windows-release-out.log](https://github.com/user-attachments/files/17176195/config-x64-windows-release-out.log)

Build works fine with the https://github.com/microsoft/vcpkg-tool/releases/tag/2024-08-01 release

**Comment by Neumann-A** [comment]
```
E:\vcpkg_folders\vcpkg-tool-bisect>git bisect start
status: waiting for both good and bad commits

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect good fd884a0d390d12783076341bd43d77c3a6a15658
status: waiting for bad commit, 1 good commit known

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect bad 5f5da308dd1dd3cc49641c60eafff176e248a9dd
Bisecting: 8 revisions left to test after this (roughly 3 steps)
[b8b3c1b31e99343cf7685aa25aaa5d2c52e33c8d] Make comparison of installed files case-insensitive (#1483)

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect good
Bisecting: 4 revisions left to test after this (roughly 2 steps)
[a65b710faf24541f51eb46eebfb22ae8079dd58f] Preserve the full exit code range in the system.process layer. (#1489)

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect bad
Bisecting: 1 revision left to test after this (roughly 1 step)
[d3fc35774f86782f356ef0f9352ba857fa57302e] Keep all CUDA_PATH_* environment variables when forming the clean environment on Windows (#1476)

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[e720ee8b6d4e3353f0efc5d88b1224320fd6f398] [upgrade] Don't crash on non-installed ports (#1473)

E:\vcpkg_folders\vcpkg-tool-bisect>git bisect good
d3fc35774f86782f356ef0f9352ba857fa57302e is the first bad commit
commit d3fc35774f86782f356ef0f9352ba857fa57302e
Author: autoantwort <41973254+autoantwort@users.noreply.github.com>
Date:   Tue Sep 3 21:14:49 2024 +0200

    Keep all CUDA_PATH_* environment variables when forming the clean environment on Windows (#1476)

 include/vcpkg/base/system.h       |  2 +-
 src/vcpkg.cpp                     |  2 +-
 src/vcpkg/base/system.cpp         |  8 ++++----
 src/vcpkg/base/system.process.cpp | 33 ++++++++++++++++-----------------
 4 files changed, 22 insertions(+), 23 deletions(-)

E:\vcpkg_folders\vcpkg-tool-bisect>
```

**Comment by Neumann-A** [comment]
https://github.com/microsoft/vcpkg-tool/pull/1476 broke it

**Comment by dg0yt** [comment]
Already noticed before the release but couldn't stop it: https://github.com/microsoft/vcpkg/issues/41199#issuecomment-2378351021

**Comment by JonLiu1993** [comment]
Fixed by PR #41271
